packages/task/src/browser/task-service.ts

```diff
@@ -974,26 +978,30 @@ export class TaskService implements TaskConfigurationClient {
         const runningTasks: TaskInfo[] = await this.getRunningTasks();
         // Get the corresponding task information based on task id if available.
         const taskInfo: TaskInfo | undefined = runningTasks.find((t: TaskInfo) => t.taskId === taskId);
-        // Create terminal widget to display an execution output of a task that was launched as a command inside a shell.
-        const widget = <TerminalWidget>await this.widgetManager.getOrCreateWidget(
-            TERMINAL_WIDGET_FACTORY_ID,
-            <TerminalWidgetFactoryOptions>{
+        let widgetOpenMode: WidgetOpenMode = 'open';
+        if (taskInfo && taskInfo.config.presentation && taskInfo.config.presentation.reveal === RevealKind.Always) {
+            if (taskInfo.config.presentation.focus) { // assign focus to the terminal if presentation.focus is true
+                widgetOpenMode = 'activate';
+            } else { // show the terminal but not assign focus
+                widgetOpenMode = 'reveal';
+            }
+        }
+
+        // Create / find a terminal widget to display an execution output of a task that was launched as a command inside a shell.
+        const widget = await this.taskTerminalWidgetManager.open(
```

Comment:
Please file an issue to separate identity from opener options for terminals and task terminals in particularly. We put too much stuff into the identity options, i.e.  `widgetOptions`, `mode`, `taskConfig` should not be stored. Local storage is not big just 5 mb for all origin data.

Reply:
i will create the GH issue after this PR gets merged.